api/forms.py

```diff
@@ -24,3 +25,17 @@ class SetAttendanceStatusForm(forms.Form):
     student_id = forms.IntegerField(label="Студент")
     attendance_id = forms.IntegerField(label="Записка")
     attendance_type = forms.IntegerField(label="Статус")
+
+
+class DutyForm(forms.Form):
+    student_id = forms.IntegerField(label="Студент")
+    comment = forms.CharField(label='Комментарий', max_length=100)
+    attendance_id = forms.IntegerField(label='День посещения')
+    duty_type = forms.CharField(label="Статус")
+    mark = forms.IntegerField(label="Оценка")
+
+    def clean_mark(self):
```

Comment:
где это используется?

Reply:
Валидатор на оценку. Но также добавили валидацию на фронт

Reply:
я понял, ткни меня в место где он вызываеся

Reply:
> The clean_<fieldname>() method is called on a form subclass – where <fieldname> is replaced with the name of the form field attribute. This method does any cleaning that is specific to that particular attribute, unrelated to the type of field that it is. This method is not passed any parameters. You will need to look up the value of the field in self.cleaned_data and remember that it will be a Python object at this point, not the original string submitted in the form (it will be in cleaned_data because the general field clean() method, above, has already cleaned the data once).
https://docs.djangoproject.com/en/2.2/ref/forms/validation/

Reply:
ок, я не знал, видел только clean()